### ISSUE TITLE
Difficulties with the autocalibrate process 

### ISSUE THREAD

**Issue by gitRonang** [fbergama/wass#27]
I'm having trouble with the auto-calibrate part and therefore rectifying my images in WASS. The extrinsic parameters and therefore the H homography matrix are causing my image to rotate more than necessary. When I calculate ext_T and ext_R separately, I get the right rotations and translations overall, but I can't find this in WASS.
Why is this? My images are probably a few mm off axis. 
What I wan't:
![stereo](https://github.com/fbergama/wass/assets/96581375/30d8ebe5-7930-4565-93e5-2063c265a3fe)
What I have with WASS autocalibrate parameters:
![stereo](https://github.com/fbergama/wass/assets/96581375/aecd17b9-465f-4ee9-a21a-7097583fc615)

Thank's!

**Comment by fbergama** [comment]
Your cameras are quite close to the water's surface. This is a tricky situation since the reflections won't match that well between the two images. 
Can you provide more info on the camera geometry? Are the two cameras mounted parallel and side by side? How did you obtain the first image?

**Comment by gitRonang** [comment]
Yes they are between 1.5 and 2m above the water surface...I could try a better angle next time!
This is my setup:
![IMG_7457](https://github.com/fbergama/wass/assets/96581375/0333cb2c-729b-4a7a-bfad-e94add328464)
![IMG_7456](https://github.com/fbergama/wass/assets/96581375/55a044df-7136-468d-a347-c0ad53617216)

Basically, 0.4m baseline, 2456*2048 images, 5 fps, More or less the totally parallel :) 
This is a first setup, I'm considering building a more robust and precise one since WASS works pretty well!

The first image: I have computed intrinsic and extrinsic parameters with an external script, with a chessboard. Then, before the stereo step I replaced the computed extrinsic parameters from WASS by mine. However, it looks good but the homography matrix don't match this parameters and the stereo step fails. I currently try to obtain my own homography matrix that I can control and then reinject it before the stereo step.

**Comment by fbergama** [comment]
Nice setup! I like it 👍 

Ok, the geometry is fine. Considering how close they are I suggest to stereo-calibrate using a chessboard as you did. The way to use it is to create two files: `ext_T.xml` and `ext_R.xml` and place them alongside the intrinsic and distortion matrices in the `config/` folder. `wass_prepare` will automatically use them. Then, skip the match and autocalibrate steps.

One last question. How do you ensure that the images are captured at the same time? Those look like rolling-shutter cameras...

**Comment by gitRonang** [comment]
Ok I'm gonna try this, thank's.

Well, I have tested multiple ways of doing the synchro between two basic camera (Arducam low distortion camera).
I achieve <1ms desynchro with a script that purge the buffer before saving the picture. A python script launch 2 calls to get camera images, purge the buffer and then accept. After 10 pictures I achieve strict synchro.

For my setup, do you think that a short baseline like that would be problematic to efficiently run WASS?

**Comment by gitRonang** [comment]
Moreover, I tried your solution with extrinsic parameters. And then I face a case that I have sometimes: 
![area_grid](https://github.com/fbergama/wass/assets/96581375/13926363-cb8c-406a-8c75-1d12040835f8)

Usually the cloud point should be inversed, maybe this is caused by these few outliers?

**Comment by gitRonang** [comment]
> Moreover, I tried your solution with extrinsic parameters. And then I face a case that I have sometimes: ![area_grid](https://user-images.githubusercontent.com/96581375/284884929-13926363-cb8c-406a-8c75-1d12040835f8.png)
> 
> Usually the cloud point should be inversed, maybe this is caused by these few outliers?

Looks like an issue with the file planes.txt...

**Comment by fbergama** [comment]
That's clearly wrong, not sure if it's the mean sea-plane estimation or a problem with the extrinsic calibration. If you put your own ext_T values, please ensure that the length of the translation vector is 1.
